osu.Game/Extensions/ContainerExtensions.cs

```diff
@@ -0,0 +1,215 @@
+    {
+        /// <summary>
+        /// Prevents this container's children from transforming when the container transforms.
+        /// </summary>
+        /// <param name="target">The target to prevent its children from transforming.</param>
+        /// <param name="parent">The parent of the target.</param>
+        /// <param name="previousParentRotation">The parent's rotation in the previous update cycle.</param>
+        /// <param name="previousParentScale">The parent's scale in the previous update cycle.</param>
+        public static void KeepChildrenUpright(this Container target, Container parent, float previousParentRotation, Vector2 previousParentScale)
+        {
+            float parentRotation = parent.Rotation;
+            Vector2 parentScale = parent.Scale;
+
+            foreach (Drawable child in target)
+            {
+                child.RotateTo(-parentRotation);
+
+                int parentRotationInQuarterTurns = (int)Math.Floor(parentRotation / 90);
+                if (parentRotationInQuarterTurns % 2 != 0)
+                    child.ScaleTo(new Vector2(parentScale.Y, parentScale.X));
+                else
+                    child.ScaleTo(parentScale);
+
+                int previousParentRotationInQuarterTurns = (int)Math.Floor(previousParentRotation / 90);
+                int rotationsNeeded = parentRotationInQuarterTurns - previousParentRotationInQuarterTurns;
+                if (rotationsNeeded < 0)
+                    rotationsNeeded += 4;
+
+                switch (rotationsNeeded)
+                {
+                    // Rotated by 90
+                    case 1:
+                        child.Origin = rotateAnchor90(child.Origin, true);
+                        break;
+
+                    // Rotated by 180
+                    case 2:
+                        child.Origin = flipAnchorXY(child.Origin);
+                        break;
+
+                    // Rotated by 270
+                    case 3:
+                        child.Origin = rotateAnchor90(child.Origin, false);
+                        break;
+
+                    default:
+                        break;
+                }
+
+                if (Math.Sign(parentScale.X) != Math.Sign(previousParentScale.X))
+                {
+                    if (parentRotationInQuarterTurns % 2 != 0)
+                        child.Origin = flipAnchorY(child.Origin);
+                    else
+                        child.Origin = flipAnchorX(child.Origin);
+                }
+
+                if (Math.Sign(parentScale.Y) != Math.Sign(previousParentScale.Y))
+                {
+                    if (parentRotationInQuarterTurns % 2 != 0)
+                        child.Origin = flipAnchorX(child.Origin);
+                    else
+                        child.Origin = flipAnchorY(child.Origin);
+                }
+            }
+        }
+
+        private static Anchor rotateAnchor90(Anchor anchor, bool clockwise)
```

Comment:
Alternative shorter implementation (barely tested, but I believe it should work):

```csharp
private static Anchor rotateAnchor90(Anchor anchor, bool clockwise)
{
    Anchor result = 0;

    if (anchor.HasFlagFast(Anchor.x0)) result |= Anchor.y0;
    if (anchor.HasFlagFast(Anchor.x1)) result |= Anchor.y1;
    if (anchor.HasFlagFast(Anchor.x2)) result |= Anchor.y2;

    if (anchor.HasFlagFast(Anchor.y0)) result |= Anchor.x2;
    if (anchor.HasFlagFast(Anchor.y1)) result |= Anchor.x1;
    if (anchor.HasFlagFast(Anchor.y2)) result |= Anchor.x0;

    return clockwise ? result : flipAnchorXY(result);
}
```

Not sure if better, but just putting it out there.